flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/Executing.java

```diff
@@ -39,14 +42,22 @@
 import javax.annotation.Nullable;
 
 import java.time.Duration;
+import java.time.Instant;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
 
 /** State which represents a running job with an {@link ExecutionGraph} and assigned slots. */
 class Executing extends StateWithExecutionGraph implements ResourceListener {
 
     private final Context context;
+    private Instant lastRescale = Instant.EPOCH;
+    private final ScheduledExecutorService scheduledExecutor = Executors.newScheduledThreadPool(1);
```

Comment:
Should we create a new thread pool or use a old thread pool?

If we must create a new thread pool, it's better to add a thread factory.

Reply:
:+1: yes I could use Context.componentMainThreadExecutor

Reply:
I also changed the tests: I no more schedule (no time based test to avoid race conditions) but rather set/check the rescale status